### ISSUE TITLE
Window is Blank/Never Loads

### ISSUE THREAD

**Issue by Shadowmech88** [slatinsky/DiscordChatExporter-frontend#17]
## Expected Behavior

To see something in the DCEF window.

## Actual Behavior

DCEF window is just solid white.

## Steps to reproduce the problem

  1. Download latest release.
  2. Run dcef.exe.
  3. See nothing.

### Environment

  - Operating system: Windows 10 version 1809 build 17763.4131
  - CPU Architecture: x86_64 (PC)
  - Flavor: Windows binary release

### DiscordChatExporter-frontend

  - [Release](https://github.com/slatinsky/DiscordChatExporter-frontend/releases) version or commit hash: v2.1.0
  - Can the issue be reproduced in the [demo](https://dcef.slada.sk/)?: No

### [DiscordChatExporter](https://github.com/Tyrrrz/DiscordChatExporter)

- Version (CLI `--version`): v2.39.1

How did you export the data?
  - Irrelevant, this issue happens even when the only guild in the exports folder is the example guild.

  

## Logs

Is there any relevant information in logs? If so, please provide them:

dcef/logs.txt: [logs.txt](https://github.com/slatinsky/DiscordChatExporter-frontend/files/11234719/logs.txt)
The backend error log was empty.

I don't really know what the problem could be. I read through the logs; nothing stuck out to me.

**Comment by slatinsky** [comment]
Hello,

Thanks for the repport

I haven't found a way to reproduce this issue, but 3 different people have reported it to me so far.

Till I find a way to reproduce this issue, you can use an older release, before the code was rewritten almost from scratch:
[Release v1.13.2](https://github.com/slatinsky/DiscordChatExporter-frontend/releases/tag/v1.13.2)

**Comment by Shadowmech88** [comment]
Will the older release work with exports generated by newer versions of Discord Chat Exporter?

**Comment by Shadowmech88** [comment]
When using the old version, how do I shut down the program? I notice that even after closing the browser and nginx window, I can then reopen the browser and go to the same URL and the frontend is still running.
Is there some process I have to kill, or will it quit after a while of receiving no requests, or what?

**Comment by slatinsky** [comment]
In the v1 release, is is just nginx that is "hosting" static files - everything is precomputed into static json files. All you need it is to kill nginx.exe

**Comment by Shadowmech88** [comment]
How is it that I can still access the page after killing nginx, though?

**Comment by slatinsky** [comment]
True, you need to close nginx manually in the latest v1 release. But just closing the console window named "nginx" should be enough.

**Comment by superstreber3** [comment]
Just wanted to add that I noticed the same issue.

**Comment by slatinsky** [comment]
According to your logs, nginx reverse proxy didn't start at all. That confirms why your backend error log was empty. The same issue as #7.

**Comment by Shadowmech88** [comment]
Anything else I could try on my end that would help you diagnose the root cause?

**Comment by Glitch3dPenguin** [comment]
I thought I was running into this same issue. However, while I was looking for this the export just suddenly worked. It takes FOREVER to load and the all of a sudden it just fired right up. If there is any information from somewhere I can provide to help I will.

**Comment by dunc07** [comment]
In case this issue is still monitored: I ran into the same issue, and figured out the cause (at least HERE, not claiming that this is the same for everyone).

I run DCEF on Linux, exposing it to the web via ngnix+SSL, as described here:
[https://github.com/slatinsky/DiscordChatExporter-frontend#5-secure-dcef](https://github.com/slatinsky/DiscordChatExporter-frontend#5-secure-dcef)

_However_, instead of forwarding **"/"**, I forwarded **"/dcef"**. And it turns out that this causes the *.js files in "frontend/assets" not to be found, since they are referenced in index.html with a leading "/". So basically:

Forwarding to "/" and hence calling...
_https://<myserver>/_
...works, as the index.html will try to load _https://<myserver>/assets/whatever.js_, which is just fine.

Forwarding to "/dcef" and hence calling...
_https://<myserver>/dcef_
...will NOT work, as the index.html will try to load _https://<myserver>/assets/whatever.js_, instead of the (now required) _https://<myserver>/dcef/assets/whatever.js_.

Not sure if that leading "/" is used anywhere else in the DCEF HTML/JS sources, but for now, the only way to make it work without making the URL the default server URL globally is to use a virtual host name instead of a "subdirectory".
